javascript/U4/hw4.js

```diff
@@ -1,65 +1,23 @@
-// class MyClass {
-
-// }
-
-// const Singleton = (function () {
-//     let instance;
-
-//     function createInstance() {
-//         let classInstance = new MyClass();
-//         return classInstance;
-//     }
-
-//     return {
-//         getInstance: function () {
-//             if (!instance) {
-//                 instance = createInstance;
-//             }
-//             return instance;
-//         }
-//     }
-// })();
-
-// module.export = Singleton;
-
-// ES5
-// var Singleton = (function () {
-//     var instance = null;
-
-//     function Singleton() {
-//         if (!instance) {
-//             instance = this;
-//             this.obj = {
-//                 version: Math.floor(Math.random() * 2000)
-//             }
-//         }
-//         return instance;
-//     }
-
-//     return Singleton;
-// })();
-
-// Singleton.prototype.get = function (key) {
-//     return this.obj[key];
-// }
+//ES5
+function singleton(name) {
+    if (!singleton.instance) {
+        this.name = name
+        singleton.instance = this
+    }
+    return singleton.instance
+}
 
-// ES6 / 2015
-class Singleton {
-    constructor() {
-        if (Singleton.instance instanceof Singleton) {
-            return Singleton.instance;
+//ES6
+class singleton {
+    constructor(name) {
+        if (!singleton.instance) {
```

Comment:
you have to check if the instance is one of Singleton's instance, otherwise there could be a glitch. E.g. if i set `Singleton.instance = 'test'`, Singleton class is not able to create any new instance at all